### PR TITLE
Feature/#191 add versioning to tour

### LINKED ISSUE #190: Add versioning feature to sub-techniques
As a user of the ATT&CK website, I want to see the new ATT&CK versions feature implemented on the sub-techniques branch.

### PR DESCRIPTION
**Pull request by IanDavila**
## Description of what has changed

Added versioning feature. Modified matrix header layout. Added versioning feature to tour.

## Issues addressed by pull request

 

Closes #190 and #191.

### CODE REVIEW

File: attack-theme/templates/macros/matrix.html [hunk 1]
```diff
@@ -89,6 +88,7 @@ <h3 class="text-center matrix-title">{{matrix.name}}</h3>
     </div>
     {% if not isIndex %}
     <div class="timestamp deemphasis text-center pt-3">Last modified: {{matrix.timestamp}}</div>
+    {% set permalink_title = "the matrices" if matrix.tactics | length > 1 else "the matrix" %}
```
Comment (isaisabel) [inline attack-theme/templates/macros/matrix.html:91]: I think this is dead code from an earlier implementation, correct?

File: data/versions.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{
+    "current": {
+        "name": "v6.3",
+        "date_start": "October 24, 2019",
+        "changelog": "updates-october-2019",
+        "cti_url": "https://github.com/mitre/cti/releases/tag/ATT%26CK-v6.3"
+    },
```
Comment (isaisabel) [inline data/versions.json:7]: We should stage this as a previous version for sub-techniques 2.1, and make the current version `ATT&CK-v7.0` (which doesn't technically exist yet, but will once we release the sub-techniques update). sub-techniques 2.1 isn't being released until the actual update on the 8th.

```
        {
            "name": "v6.3",
            "aliases": [],
            "date_start": "October 24, 2019",
            "date_end": "July 7, 2020",
            "changelog": "updates-october-2019",
            "cti_url": "https://github.com/mitre/cti/releases/tag/ATT%26CK-v6.3",
            "commit": "b15db5ce0b2f440cb690a3003ffc689098625d00"
        },
```

Please also modify the versions script remove the `/beta/` directory, or else we're going to preserve the sub-techniques beta with v6.3. We may have other betas in the future so this is actually useful long term.

File: attack-theme/templates/macros/matrix.html (hunk 1 above)
Comment (IanDavila) [reply attack-theme/templates/macros/matrix.html:91]: I don't think that it is dead code. It's used for the tooltip of the Version Permalink button.

Comment (isaisabel) [reply attack-theme/templates/macros/matrix.html:91]: `permalink_title` only occurs once in the matrix macro: https://github.com/mitre-attack/attack-website/blob/feature/%23191-add-versioning-to-tour/attack-theme/templates/macros/matrix.html#L91

File: data/versions.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    "previous": [
+        {
+            "name": "v6.3",
+            "aliases": [],
+            "date_start": "October 24, 2019",
+            "date_end": "July 7, 2020",
+            "changelog": "updates-october-2019",
+            "cti_url": "https://github.com/mitre/cti/releases/tag/ATT%26CK-v6.3",
+            "commit": "b15db5ce0b2f440cb690a3003ffc689098625d00"
```
Comment (isaisabel) [inline data/versions.json:16]: This is actually a TODO for me, but we need to change all Navigator links to use the old Navigator. [I still need to stand up an instance](https://github.com/mitre-attack/attack-navigator/issues/143) but then I'll branch of gh-pages and do a global replace for the navigator link to target the preserved previous version.

Comment (isaisabel) [reply data/versions.json:16]: To be clear, this applies to the v6.3 version of the website, not the v7.0 version.

**Pull request merged**